### ISSUE TITLE
linter: `unicorn/prefer-at` misses single-character `substring()` access

### ISSUE THREAD

**Issue by lukpsaxo** [oxc-project/oxc#26595]
### What version of Oxlint are you using?

1.82.0

### What command did you run?

oxlint

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```json
{
    "plugins": ["unicorn"],
    "rules": {
        "unicorn/prefer-at": "error"
    }
}
```

### What happened?

Oxlint 1.82.0 does not report `String#substring(index, index + 1)`. `eslint-plugin-unicorn` 74.0.0 reports this as `unicorn/prefer-at` and suggests `String#at(index)`.

```js
value.substring(index, index + 1);
```

## Expected

One `unicorn/prefer-at` error, equivalent to:

> Prefer `String#at(…)` over `String#substring(…)` when getting one character.

## Actual

No `unicorn/prefer-at` error.
